ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -315,10 +315,14 @@ protected Boolean findNextCompactionAndExecute(boolean collectGenericStats, bool
         return false;
       }
 
-      if (!ci.type.equals(CompactionType.REBALANCE) && ci.numberOfBuckets > 0) {
-        if (LOG.isWarnEnabled()) {
-          LOG.warn("Only the REBALANCE compaction accepts the number of buckets clause (CLUSTERED INTO {N} BUCKETS). " +
-              "Since the compaction request is {}, it will be ignored.", ci.type);
+      boolean insertOnly = AcidUtils.isInsertOnlyTable(table.getParameters());
+      if (LOG.isWarnEnabled() && ci.type.equals(CompactionType.REBALANCE) && insertOnly) {
```

Comment:
Can `ci` and `ci.type` be null?

Reply:
No, ci is checked at line 273
```
      if (ci == null) {
        return false;
      }
```
type checked at org.apache.hadoop.hive.metastore.txn.TxnUtils#dbCompactionType2ThriftType